### PR TITLE
Fix #301, Add Github workflows to bootes.x

### LINKED ISSUE #301: Port Codeql and workflows into bootes
**Checklist (Please check before submitting)**

- [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
- [x] I reviewed the [cFS README.md file](https://github.com/nasa/cFS/blob/main/README.md) to see if the feature is in the major future work.
- [x] I performed a cursory search to see if the feature request is relevant, not redundant, nor in conflict with other tickets.

**Is your feature request related to a problem? Please describe.**
The current build of bootes lack the Codeql  semantic code analyzer and GitHub action workflows.

**Describe the solution you'd like**
Add workflows and Codeql to the bootes build. 

**Describe alternatives you've considered**
None

**Requester Info**
Frank I. Gonzalez Soto -GSFC 582 Intern

### PR DESCRIPTION
**Pull request by Frank393**
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [ ] I signed and emailed the appropriate Contributor License agreement to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
- Redo of pull request #305 
- Added workflow directory and Codeql files for the Bootes build to fixes #301  .
- Modified  build-documentation.yml and static-analysis.yml to work with Bootes build.
- Modified the workflow README.md badges to work with bootes.x branch.

**Testing performed**
Steps taken to test the contribution:
1. Modified path for the build-documentation.yml build OSAL Guide so it could find the correct directory.
Documentation workflow [run](https://github.com/Frank393/cFS/actions/runs/1037623337)
2.  Modified path for the static-analysis.yml cfe strict cppcheck to find the correct directory.
Static analysis workflow [run](https://github.com/Frank393/cFS/actions/runs/1037623341)

**Expected behavior changes**

- With the added workflow and Codeql we can test the previous build with GitHub action and Codeql semantic code analyzer.

**System(s) tested on**
 - GitHub actions ubuntu-18.04 runner

**Contributor Info - All information REQUIRED for consideration of pull request**
Frank I. Gonzalez Soto , GSFC Code 582 intern

### CODE REVIEW

File: .github/workflows/build-cfs-deprecated.yml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    # Set the type of machine to run on
+    env:
+      BUILDTYPE: ${{ matrix.buildtype }}
+
+    steps:
+      # Checks out a copy of your repository on the ubuntu-latest machine
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
```
Comment (astrogeco) [inline .github/workflows/build-cfs-deprecated.yml:50]: This should checkout the bootes branch instead of main

**Review by astrogeco — CHANGES_REQUESTED** [review]
Change the pull_request trigger to only focus on bootes.x branch. 

Change all checkout actions to checkout bootes branch instead of main

File: .github/workflows/build-cfs-deprecated.yml [hunk 2]
```diff
@@ -0,0 +1,163 @@
+name: "Deprecated Build, Test, and Run"
+
+# Run every time a new commit pushed or for pull requests
+on:
+  push:
+  pull_request:
```
[the remark this answers is not in the document]
Comment (skliper) [reply .github/workflows/build-cfs-deprecated.yml:6]: I don't follow why you'd need to specify... if the workflow is specific to the bootes.x branch already it's only going to run within that context?  It'd only be needed if you merged this workflow to other branches but only wanted to run it in certain cases... as it is you want to run it for any branch it's in right?

[the remark this answers is not in the document]
Comment (astrogeco) [reply .github/workflows/build-cfs-deprecated.yml:6]: I think you're right, I was wondering if maybe for the pull_request one? Especially if we change the checkout action to target bootes

[the remark this answers is not in the document]
Comment (skliper) [reply .github/workflows/build-cfs-deprecated.yml:6]: But wouldn't the workflow still only be in PR's that are applicable?  I haven't been able to think of a case where you wouldn't want to run the workflow (based on target branch or whatever) if it's in your branch.

**Comment by astrogeco** [comment]
[CCB:2021-08-04](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-August-04-2021) APPROVED

**Review by astrogeco — APPROVED** [review]

**Pull request merged**